sys/auto_init/saul/auto_init_adxl345.c

```diff
@@ -41,15 +41,20 @@ static adxl345_t adxl345_devs[ADXL345_NUM];
  */
 static saul_reg_t saul_entries[ADXL345_NUM];
 
+/**
+ * @brief   Define the number of saul info
+ */
+#define ADXL345_INFO_NUM (sizeof(adxl345_saul_info) / sizeof(adxl345_saul_info[0]))
+
 /**
  * @brief   Reference the driver structs
- * @{
  */
 extern saul_driver_t adxl345_saul_driver;
-/** @} */
 
 void auto_init_adxl345(void)
 {
+    assert(ADXL345_INFO_NUM == ADXL345_NUM);
```

Comment:
Please consider https://github.com/RIOT-OS/RIOT/pull/8669#discussion_r172797467 before merging.

Reply:
Not possible (see https://github.com/RIOT-OS/RIOT/pull/8669#discussion_r172821505) so let's rebuild and then go.